packages/pn-pa-webapp/src/pages/components/ApiKeys/DesktopApiKeys.tsx

```diff
@@ -185,23 +189,27 @@ const DesktopApiKeys = ({
     },
   ];
 
-  const rows: Array<Item> = apiKeys.map((n: ApiKey, index) => ({
-    ...n,
-    id: index.toString(),
-  }));
+  useEffect(() => {
+    const rowsMap: Array<Item> = apiKeys.map((n: ApiKey, index) => ({
+      ...n,
+      id: index.toString(),
+    }));
+    setRows(rowsMap);
+    setTableKey(tableKey + 1);
+  }, [apiKeys]);
 
   return (
     <Fragment>
       {apiKeys && (
         <Fragment>
           {apiKeys.length > 0 ? (
-            <ItemsTable data-testid="tableApiKeys" columns={columns} rows={rows} />
+            <ItemsTable key={tableKey} data-testid="tableApiKeys" columns={columns} rows={rows} />
```

Comment:
I don't understand why it is needed to assign a key to this component, moreover a key that autoincrements at each render. I imagine there should be a good reason for it. So I ask you please to add a comment regarding this detail, thanks!

Reply:
For some reason, React won't properly update components after dispatching the geApiKeys when an operation is done (deleting or setting status). The reassigning value to key property made it.